lib/chef-dk/command/generator_commands.rb

```diff
@@ -69,6 +69,12 @@ module SharedGeneratorOptions
         :description => "Use to set arbitrary attribute KEY to VALUE in the code_generator cookbook",
         :default => [],
         :proc => Proc.new { |s| generator_args <<  s.split("=").map(&:strip) }
+
+      option :branch,
```

Comment:
This is only relevant to generating build cookbooks, right? If so, it should only be added in `BuildCookbook`. Put it above the `options.merge!(SharedGeneratorOptions.options)` line.